### DISCUSSION TITLE
Alarm polling frequency

### DISCUSSION THREAD

**Discussion by rezzalex** [duanemck/ezviz.net#8] [Q&A] [answered]
Hello.

Reading the wiki, it seems we can configure the alarm polling frequency.
Does that mean it is NOT real time ?
I thought this kind of things was pushed by Ezviz on an event base, not a time base.

Thank you.

**Comment by duanemck** [comment]
Yeah, unfortunately this is a reverse-engineered library based on a Python library that did some initial work and then I expanded on that. The mobile app does have event based alarms but I haven't yet been able to figure out how to reverse engineer that. The polling was the next best thing. 

If someone is willing to dig into the mobile app and figure that out it would be a great addition.

**Reply by rezzalex** [reply]
I have first found this project, maybe this is the one you were reffering to : 
https://github.com/BaQs/pyEzviz/issues/98

on that issue, I have modified the code so that I call Domoticz APIs from the live mqtt motion detection : 

maybe you could check that  code ?

**Reply by duanemck** [reply]
Yes, that is the one I used initially but it was a few years ago and things might have been updated there.
I'll have a look at the issue and see if there's something I can pull across to this app, thank you!

**Reply by rezzalex** [reply]
in this part of the program, it seems that the programm subscribe to the official EZVIZ mqtt flow ==> real time : https://github.com/BaQs/pyEzviz/blob/master/pyezviz/mqtt.py

**Reply by duanemck** [reply]
Thanks so much, that code was very useful!

I've just pushed a `beta` tagged image to Docker hub, give that a go.
You'll also need to add the following to your `ezviz` section in the `appsettings.json`:

```
    "enablePushNotifications" : true
```

One thing to note, it looks like Ezviz only support 1 device being subscribed at a time, at least when I opened the app (or the pyEzviz script) I seemed to stop getting messages pushed to me. I've added some code to try detect this and re-subscribe but it's possible that while this is running you won't get notifications on the app anymore. (This is just a suspicion, I could be wrong, would be interested to hear your findings)

**Comment by rezzalex** [comment]
Hello @duanemck , thanks a lot, this is working

I did "arm" my cam with the mobile app.
Then I moved my hand in front of the cam while the ezviz mobile app was opened.
I received the alarm on the phone, with an email AND in MQTT explorer !!!! :

![image](https://github.com/duanemck/ezviz.net/assets/35136439/be74fc89-ac53-4ad9-9e20-44a343451356)

I am trying to understand how to retrieve the picture taken ...

**Reply by duanemck** [reply]
To download the Picture with the `PicUrl` link you will need to have a logged in session.
The image should be included in the message as `DownloadedPicture`. It's a Base64 encoded image. I used NodeRed to convert it back to an image, you could test it with something like: https://codebeautify.org/base64-to-image-converter.

**Reply by rezzalex** [reply]
Thank you.
How can I have a logged in session when https://ieu.ezvizlife.com is working only with IE and this browser is not supported anymore... ?

Ideally, I would like to save the image and or send it by email, Which is strangely not working out of the box in EZVIZ app

**Reply by duanemck** [reply]
Is there a reason you don't want to use the Base64 encoded image that's in the MQTT message?

If you really want to download it yourself, this is how you authenticate:

You need to POST the following payload to `https://apiieu.ezvizlife.com/v3/users/login/v5`

```
{
   "account": "<<YOUR USERNAME>>" ,
   "password": "<<YOUR_PASSWORD>>" ,
   "cuName": "SFRDIDEw" ,
   "msgType": "0" ,
   "feature_code": "1fc28fa018178a1cd1c091b13b2f9f02",
}
```

You'll get something like this back:

```
{
  //a bunch of other fields,
  "loginSession: {
    "sessionId": "<<SESSION ID>>"
  }
}
```

Then when you download the image, you need to set a header called `sessionId` with the above `<<SESSION ID>>`.

**Reply by rezzalex** [reply]
Thank you @duanemck 

My goal is to receive by email a cam snapshot on motion detection event.
Ezviz propose natively to send an email, but this one does not contain any picture,
you need to go on their mobile app to get the picture, I would like this kind of notification to be more "pushy"

**Comment by rezzalex** [comment, marked as answer]
Would it be possible to push each cam and global status as well?

**Reply by duanemck** [reply]
I've only seen Motion and Offline alarms coming through the push channel from ezviz. I guess it's up to them as to what they send, I don't know of any way to register for specific events.

**Reply by rezzalex** [reply]
thank you, maybe low just battery notification, but if we can poll it manually, which is the case, the benefit of a real time push is not obvious...
